### PR TITLE
feat: migrate arc & paraswap

### PR DESCRIPTION
**Pull request by sakulstra**

### CODE REVIEW

File: src/20240130_AaveV2Ethereum_MigrationOfRemainingGovV2Permissions/AaveV2Ethereum_MigrationOfRemainingGovV2Permissions_20240130.t.sol [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  }
+
+  /**
+   * @dev executes the generic test suite including e2e and config snapshots
+   */
+  function test_defaultProposalExecution() public {
+    vm.warp(block.timestamp + estimatedExecutionDelay);
+    executePayload(vm, address(proposal));
+    vm.warp(block.timestamp + 1 days);
+    executePayload(vm, address(proposalPart2));
```
Comment (sendra) [inline src/20240130_AaveV2Ethereum_MigrationOfRemainingGovV2Permissions/AaveV2Ethereum_MigrationOfRemainingGovV2Permissions_20240130.t.sol:42]: would put here a comment maybe to add specific payload tests

**Review by sendra — DISMISSED** [review]

File: src/20240130_AaveV2Ethereum_MigrationOfRemainingGovV2Permissions/AaveV2Ethereum_MigrationOfRemainingGovV2Permissions_20240130.sol [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+  function getDelay() external view returns (uint256);
+}
+
+/**
+ * @title Migration of remaining gov v2 executor controlled systems
+ * - migrating paraswap collector funds
+ * - migrating arc permissions
+ * @author BGD Labs @bgdlabs
+ * - Snapshot: TODO
```
Comment (sakulstra) [inline src/20240130_AaveV2Ethereum_MigrationOfRemainingGovV2Permissions/AaveV2Ethereum_MigrationOfRemainingGovV2Permissions_20240130.sol:34]:
```suggestion
 * - Snapshot: https://snapshot.org/#/aave.eth/proposal/0x4fd357741900bfe62a863d1e3ec84fbf79bfebd5bdda3f66eee75b8845274b6d
```

File: src/20240130_AaveV2Ethereum_MigrationOfRemainingGovV2Permissions/MigrationOfRemainingGovV2Permissions.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+On mainnet the proposal also queues a call to:
+
+- `arcTimelock.updateEthereumGovernanceExecutor(GovernanceV3Ethereum.EXECUTOR_LVL_1)`
+
+## References
+
+- Implementation: [AaveV2Ethereum](https://github.com/bgd-labs/aave-proposals-v3/blob/main/src/20240130_AaveV2Ethereum_MigrationOfRemainingGovV2Permissions/AaveV2Ethereum_MigrationOfRemainingGovV2Permissions_20240130.sol)
+- Tests: [AaveV2Ethereum](https://github.com/bgd-labs/aave-proposals-v3/blob/main/src/20240130_AaveV2Ethereum_MigrationOfRemainingGovV2Permissions/AaveV2Ethereum_MigrationOfRemainingGovV2Permissions_20240130.t.sol)
+- [Snapshot](TODO)
```
Comment (sakulstra) [inline src/20240130_AaveV2Ethereum_MigrationOfRemainingGovV2Permissions/MigrationOfRemainingGovV2Permissions.md:32]:
```suggestion
- [Snapshot](https://snapshot.org/#/aave.eth/proposal/0x4fd357741900bfe62a863d1e3ec84fbf79bfebd5bdda3f66eee75b8845274b6d)
```

File: src/20240130_AaveV2Ethereum_MigrationOfRemainingGovV2Permissions/MigrationOfRemainingGovV2Permissions.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
+---
+title: "Migration of remaining gov v2 permissions"
```
Comment (sakulstra) [inline src/20240130_AaveV2Ethereum_MigrationOfRemainingGovV2Permissions/MigrationOfRemainingGovV2Permissions.md:2]:
```suggestion
title: "Migration of arc permissions & old paraswap claims"
```

File: src/20240130_AaveV2Ethereum_MigrationOfRemainingGovV2Permissions/MigrationOfRemainingGovV2Permissions.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+## Simple Summary
+
+Migrate aave arc pool permissions & funds allocated to the old governance v2 short executor.
+
+## Motivation
+
+In November 2022 a permissionless contract was [introduced](https://governance.aave.com/t/bgd-aave-paraswap-fee-claimer/10671) to collect positive slippage from paraswap swaps to the aave-collector. While this system is well and active since [then](https://dashboard.paraswap.io/public/dashboard/5b6dae52-b39e-4c49-a670-e0f0c0aebee5?partner=aave) there are some funds(~100k) stuck in the previous system which still need [migration](https://governance.aave.com/t/bgd-aave-paraswap-fee-claimer/10671/3).
+
+When governance v3 was introduced, some permissions for the deprecated aave arc pool were not properly migrated to the new governance system. For proper permission alignment this should still be done.
+
```
Comment (sakulstra) [inline src/20240130_AaveV2Ethereum_MigrationOfRemainingGovV2Permissions/MigrationOfRemainingGovV2Permissions.md:17]:
```suggestion

Approving this proposal will also authorize the guardians on `optimism`, `arbitrum`, `avalanche` and `fantom` to claim the pending paraswap positive slippage to the respective network collector.

```

**Review by sendra — APPROVED** [review]

**Review by kyzia551 — APPROVED** [review]

**Pull request merged**